### PR TITLE
Contribute buildah.spec

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

**Comment by rhatdan** [comment]
Currently this is blowing up.  Not sure why

```
rpmbuild -ba buildah.spec
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.Jnnk34
+ umask 022
+ cd /home/dwalsh/rpmbuild/BUILD
+ cd /home/dwalsh/rpmbuild/BUILD
+ rm -rf buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5
+ /usr/bin/gzip -dc /home/dwalsh/rpmbuild/SOURCES/buildah-6d8bf00.tar.gz
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ /usr/bin/git init -q
+ /usr/bin/git config user.name rpm-build
+ /usr/bin/git config user.email '<rpm-build>'
+ /usr/bin/git add .
+ /usr/bin/git commit -q -a --author 'rpm-build <rpm-build>' -m 'buildah-0.1 base'
+ sed -i 's/$(GO) install/$(GO) build/g' Makefile
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.53OmP0
+ umask 022
+ cd /home/dwalsh/rpmbuild/BUILD
+ cd buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5
+ mkdir _build
+ pushd _build
~/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5/_build ~/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5
+ mkdir -p src/github.com/projectatomic
++ dirs +1 -l
+ ln -s /home/dwalsh/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5 src/github.com/projectatomic/buildah
+ popd
~/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5
+ ln -s vendor src
++ pwd
++ pwd
+ export GOPATH=/home/dwalsh/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5/_build:/home/dwalsh/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5:/usr/share/gocode
+ GOPATH=/home/dwalsh/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5/_build:/home/dwalsh/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5:/usr/share/gocode
+ make all
go build -o buildah -tags "  " ./cmd/buildah
# _/home/dwalsh/rpmbuild/BUILD/buildah-6d8bf008d70c6e1f4e3fca76a172ccc7b00c0ac5/cmd/buildah
cmd/buildah/bud.go:168: cannot use "github.com/containers/storage/pkg/archive".Gzip (type "github.com/containers/storage/pkg/archive".Compression) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Compression in field value
cmd/buildah/bud.go:175: cannot use store (type "github.com/containers/storage/storage".Store) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store in argument to imagebuildah.BuildDockerfiles:
	"github.com/containers/storage/storage".Store does not implement "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store (wrong type for ApplyDiff method)
		have ApplyDiff(string, "github.com/containers/storage/pkg/archive".Reader) (int64, error)
		want ApplyDiff(string, "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Reader) (int64, error)
cmd/buildah/commit.go:78: cannot use compress (type "github.com/containers/storage/pkg/archive".Compression) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Compression in field value
cmd/buildah/commit.go:81: cannot use dest (type "github.com/containers/image/types".ImageReference) as type "github.com/projectatomic/buildah/vendor/github.com/containers/image/types".ImageReference in argument to builder.Commit:
	"github.com/containers/image/types".ImageReference does not implement "github.com/projectatomic/buildah/vendor/github.com/containers/image/types".ImageReference (wrong type for DeleteImage method)
		have DeleteImage(*"github.com/containers/image/types".SystemContext) error
		want DeleteImage(*"github.com/projectatomic/buildah/vendor/github.com/containers/image/types".SystemContext) error
cmd/buildah/common.go:40: cannot use store (type "github.com/containers/storage/storage".Store) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store in argument to buildah.OpenBuilder:
	"github.com/containers/storage/storage".Store does not implement "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store (wrong type for ApplyDiff method)
		have ApplyDiff(string, "github.com/containers/storage/pkg/archive".Reader) (int64, error)
		want ApplyDiff(string, "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Reader) (int64, error)
cmd/buildah/common.go:45: cannot use store (type "github.com/containers/storage/storage".Store) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store in argument to buildah.ImportBuilder:
	"github.com/containers/storage/storage".Store does not implement "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store (wrong type for ApplyDiff method)
		have ApplyDiff(string, "github.com/containers/storage/pkg/archive".Reader) (int64, error)
		want ApplyDiff(string, "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Reader) (int64, error)
cmd/buildah/common.go:58: cannot use store (type "github.com/containers/storage/storage".Store) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store in argument to buildah.OpenAllBuilders:
	"github.com/containers/storage/storage".Store does not implement "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store (wrong type for ApplyDiff method)
		have ApplyDiff(string, "github.com/containers/storage/pkg/archive".Reader) (int64, error)
		want ApplyDiff(string, "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Reader) (int64, error)
cmd/buildah/from.go:108: cannot use store (type "github.com/containers/storage/storage".Store) as type "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store in argument to buildah.NewBuilder:
	"github.com/containers/storage/storage".Store does not implement "github.com/projectatomic/buildah/vendor/github.com/containers/storage/storage".Store (wrong type for ApplyDiff method)
		have ApplyDiff(string, "github.com/containers/storage/pkg/archive".Reader) (int64, error)
		want ApplyDiff(string, "github.com/projectatomic/buildah/vendor/github.com/containers/storage/pkg/archive".Reader) (int64, error)
make: *** [Makefile:10: buildah] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.53OmP0 (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.53OmP0 (%build)
```

**Comment by rhatdan** [comment]
@nalind PTAL

**Comment by nalind** [comment]
Symlink shenanigans don't mix well with new-style vendoring.  If you can't avoid having symlinks in the elements of $GOPATH, replacing `ln -s vendor src` with `mv vendor src` will basically turn the current setup into old-style vendoring.

**Comment by rhatdan** [comment]
Ok this spec file can now build an installable image.

**Comment by rhatdan** [comment]
I think we can consider merging this now and then start the Fedora Review Process.

**Comment by rhatdan** [comment]
@nalind PTAL

File: contrib/rpm/buildah.spec [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* create a working container, either from scratch or using an image as a starting point
+* mount a working container's root filesystem for manipulation
+* unmount a working container's root filesystem
+* use the updated contents of a container's root filesystem as a filesystem layer to create a new image
+* delete a working container or an image
+
+
+%prep
+%autosetup -Sgit -n %{name}-%{commit}
+sed -i 's/$(GO) install/$(GO) build/g' Makefile
```
Comment (nalind) [inline contrib/rpm/buildah.spec]: This doesn't match anything in the Makefile.

File: contrib/rpm/buildah.spec [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+%global _dwz_low_mem_die_limit 0
+%else
+%global debug_package   %{nil}
+%endif
+
+%global provider        github
+%global provider_tld    com
+%global project         projectatomic
+%global repo            buildah
+# https://github.com/kubernetes-incubator/cri-o
```
Comment (nalind) [inline contrib/rpm/buildah.spec]: Copy/paste leftover?

File: contrib/rpm/buildah.spec [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path     %{provider_prefix}
+%global commit         0235b96c9b9b1c656eab50e63a0d50bb23beb8e0
+%global shortcommit    %(c=%{commit}; echo ${c:0:7})
+
+Name:           buildah
+Version:        0.1
```
Comment (nalind) [inline contrib/rpm/buildah.spec]: This doesn't jive with the output we get from "buildah --version", which is 0.0.0 until we start overriding it in main.go.

**Comment by rhatdan** [comment]
I think some of the Requires are wrong also.  Since these are cut and paste from cri-o.

**Comment by rhatdan** [comment]
Moved buildah to 0.0.1 and fixed the cut and paste error.

**Comment by rhatdan** [comment]
@nalind fixed last comment, which I missed the first time,  Sorry.

File: contrib/rpm/buildah.spec [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+License:        ASL 2.0
+URL:            https://%{provider_prefix}
+Source:         https://%{provider_prefix}/archive/%{commit}/%{repo}-%{shortcommit}.tar.gz
+
+ExclusiveArch:  x86_64 aarch64 ppc64le
+# If go_compiler is not set to 1, there is no virtual provide. Use golang instead.
+BuildRequires:  %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires:  git
+BuildRequires:  glib2-devel
+BuildRequires:  glibc-static
```
Comment (nalind) [inline contrib/rpm/buildah.spec]: Is glibc-static required?

File: contrib/rpm/buildah.spec [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+Summary:        A command line tool used to creating OCI Images
+License:        ASL 2.0
+URL:            https://%{provider_prefix}
+Source:         https://%{provider_prefix}/archive/%{commit}/%{repo}-%{shortcommit}.tar.gz
+
+ExclusiveArch:  x86_64 aarch64 ppc64le
+# If go_compiler is not set to 1, there is no virtual provide. Use golang instead.
+BuildRequires:  %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires:  git
+BuildRequires:  glib2-devel
```
Comment (nalind) [inline contrib/rpm/buildah.spec]: Is glib2-devel required?

**Comment by nalind** [comment]
LGTM.  For my future reference, are we expecting to synchronize this the version of the .spec file in [the Fedora package](http://pkgs.fedoraproject.org/cgit/rpms/buildah.git/tree/buildah.spec)?

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 2266cb0

**Pull request closed without merging**